packages/runtime/container-runtime/src/summaryGenerator.ts

```diff
@@ -14,12 +14,28 @@ import {
 } from "./summarizerTypes";
 import { IClientSummaryWatcher } from "./summaryCollection";
 
-export const checkNotTimeout = <T>(something: T | IPromiseTimerResult | undefined): something is T => {
-    if (something === undefined) {
-        return false;
-    }
-    return (something as IPromiseTimerResult).timerResult === undefined;
+export type RaceTimerResult<T> = {
+    readonly result: IPromiseTimerResult["timerResult"];
+} | {
+    readonly result: "timeoutError" | "error";
+    readonly error: any;
+} | {
+    readonly result: "done";
+    readonly value: T;
 };
+export const raceTimer = async <T>(
+    p: Promise<T>,
+    timer: Promise<IPromiseTimerResult>,
+): Promise<RaceTimerResult<T>> => Promise.race([
+    p.then(
+        (value) => ({ result: "done", value } as const),
+        (error) => ({ result: "timeoutError", error } as const),
```

Comment:
I'm a bit confused about "timeoutError". It's not timeout error, it's just an error, no?
And the error case in timer is timeoutError

Reply:
Oh I had these reversed, nice catch!